### ISSUE TITLE
how i train direct features extracted from other model using yolov8?

### ISSUE THREAD

**Issue by manishp54** [ultralytics/ultralytics#14318]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have video feature of dimension (500,810 512). i want to classify these feature into six classes. how I used that feature to train yolov8 classifier.

**Comment by Y-T-G** [comment]
Why do you need YOLO for that? Just use PyTorch.

**Comment by glenn-jocher** [comment]
@Y-T-G hello! Thanks for reaching out.

To classify features extracted from another model using YOLOv8, you can indeed leverage PyTorch for this task. YOLOv8 is primarily designed for object detection, segmentation, and pose estimation, but you can still use its underlying architecture and PyTorch capabilities to achieve your goal.

Here's a concise guide on how you can proceed:

1. **Prepare Your Dataset**: Ensure your features and labels are properly formatted. You should have a dataset where each sample is a feature vector of shape `(512,)` and a corresponding label.

2. **Define a Custom Classifier**: You can create a simple neural network classifier using PyTorch. Here's an example:

    ```python
    import torch
    import torch.nn as nn
    import torch.optim as optim
    from torch.utils.data import DataLoader, TensorDataset

    # Example feature and label tensors
    features = torch.randn(500, 512)  # 500 samples, each of dimension 512
    labels = torch.randint(0, 6, (500,))  # 500 labels for 6 classes

    # Create a dataset and dataloader
    dataset = TensorDataset(features, labels)
    dataloader = DataLoader(dataset, batch_size=32, shuffle=True)

    # Define a simple classifier
    class SimpleClassifier(nn.Module):
        def __init__(self, input_dim, num_classes):
            super(SimpleClassifier, self).__init__()
            self.fc = nn.Linear(input_dim, num_classes)

        def forward(self, x):
            return self.fc(x)

    model = SimpleClassifier(input_dim=512, num_classes=6)
    criterion = nn.CrossEntropyLoss()
    optimizer = optim.Adam(model.parameters(), lr=0.001)

    # Training loop
    for epoch in range(10):  # Train for 10 epochs
        for batch_features, batch_labels in dataloader:
            optimizer.zero_grad()
            outputs = model(batch_features)
            loss = criterion(outputs, batch_labels)
            loss.backward()
            optimizer.step()

        print(f'Epoch {epoch+1}, Loss: {loss.item()}')

    print("Training complete!")
    ```

3. **Evaluate Your Model**: After training, you can evaluate your model on a validation set to check its performance.

This approach allows you to leverage the power of PyTorch for your classification task while keeping the process straightforward and efficient. If you have any further questions or need additional assistance, feel free to ask!

For more detailed guidance on using PyTorch, you can refer to the [PyTorch documentation](https://pytorch.org/docs/stable/index.html).

Best of luck with your project! 🚀
